datasets/msr_zhen_translation_parity/msr_zhen_translation_parity.py

```diff
@@ -0,0 +1,112 @@
+
+import datasets
+
+
+_DESCRIPTION = """\
+Translator Human Parity Data
+
+Human evaluation results and translation output for the Translator Human Parity Data release, 
+as described in https://blogs.microsoft.com/ai/machine-translation-news-test-set-human-parity/. 
+The Translator Human Parity Data release contains all human evaluation results and translations
+related to our paper "Achieving Human Parity on Automatic Chinese to English News Translation", 
+published on March 14, 2018.
+"""
+
+_LICENSE = """\
+See the Microsoft Research Data License Agreement.
+"""
+
+_CITATION = """\
+@misc{hassan2018achieving,
+      title={Achieving Human Parity on Automatic Chinese to English News Translation}, 
+      author={ Hany Hassan and Anthony Aue and Chang Chen and Vishal Chowdhary and Jonathan Clark 
+               and Christian Federmann and Xuedong Huang and Marcin Junczys-Dowmunt and William Lewis 
+               and Mu Li and Shujie Liu and Tie-Yan Liu and Renqian Luo and Arul Menezes and Tao Qin 
+               and Frank Seide and Xu Tan and Fei Tian and Lijun Wu and Shuangzhi Wu and Yingce Xia 
+               and Dongdong Zhang and Zhirui Zhang and Ming Zhou},
+      year={2018},
+      eprint={1803.05567},
+      archivePrefix={arXiv},
+      primaryClass={cs.CL}
+}"""
+
+_FILE_PATHS = [
+    ["References", "Translator-HumanParityData-Reference-HT.txt"],
+    ["References", "Translator-HumanParityData-Reference-PE.txt"],
+    ["Translations", "Translator-HumanParityData-Combo-4.txt"],
+    ["Translations", "Translator-HumanParityData-Combo-5.txt"],
+    ["Translations", "Translator-HumanParityData-Combo-6.txt"],
+    ["Translations", "Translator-HumanParityData-Online-A-1710.txt"],
+]
+
+_FEATURE_NAMES = ["Reference-HT", "Reference-PE", "Combo-4", "Combo-5", "Combo-6", "Online-A-1710"]
+
+
+class MsrZhenTranslationParity(datasets.GeneratorBasedBuilder):
+    """Microsoft Research: Translator Human Parity Data"""
+
+    VERSION = datasets.Version("1.0.0")
+
+    @property
+    def manual_download_instructions(self):
+        return """\
+    You need to go to https://msropendata.com/datasets/93f9aa87-9491-45ac-81c1-6498b6be0d0b,
+    and manually download translatorhumanparitydata2.zip (or translatorhumanparitydata2.tar.gz).
+    Once it is completed, extract its content into a directory, <path/to/folder>.
+    Within this directory, there are three subdirectories, Translations, References, and Evaluations.
+    The <path/to/folder> can e.g. be "~/Downloads/translatorhumanparitydata2", if you just double click the .zip file.
+    msr_zhen_translation_parity can then be loaded using the following command 
+    `datasets.load_dataset("msr_zhen_translation_parity", data_dir="<path/to/folder>")`.
+    """
+
+    def _info(self):
+        feature_names = _FEATURE_NAMES
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features({k: datasets.Value("string") for k in feature_names}),
```

Comment:
do you think we can use the `Translation` feature type for the two `Reference` columns, since they're parallel translations if I understand correctly ?

you can see an example of usage in [flores](https://github.com/huggingface/datasets/blob/master/datasets/flores/flores.py)

Reply:
All six translations in this data are English translation of the Chinese-English language pair of WMT17. The downloaded package does not provide the source language (Chinese). I think for that we can reuse existing `wmt17` dataset within this repo. But I have not tried it yet. By this dataset itself, all six features are only the targets of a translation. I will have another look of flores. But a quick skim indicates that a `Translation` feature type requires a source language, which this dataset does not have by itself.

Reply:
Ok you're right ! Let's keep it this way then

Reply:
(especially because it looks like wmt17 has some issues right now)